ddl/constraint.go

```diff
@@ -205,29 +206,50 @@ func checkDropCheckConstraint(t *meta.Meta, job *model.Job) (*model.TableInfo, *
 	return tblInfo, constraintInfo, nil
 }
 
-func (w *worker) onAlterCheckConstraint(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error) {
+func (w *worker) onAlterCheckConstraint(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, err error) {
 	dbInfo, tblInfo, constraintInfo, enforced, err := checkAlterCheckConstraint(t, job)
 	if err != nil {
 		return ver, errors.Trace(err)
 	}
 
 	// enforced will fetch table data and check the constraint.
-	if constraintInfo.Enforced != enforced && enforced {
-		err = w.verifyRemainRecordsForCheckConstraint(dbInfo, tblInfo, constraintInfo, job)
```

Comment:
In the check constraint, when changing from `not enforced` to `enforced`, remaining data verification is required. All nodes in tidb may have two states, `enforced` and `not enforced`. After the remaining data is verified, data that violates the check constraint may also be written. Therefore, state changes are added to ensure that all nodes have reached the `enforced` state before remaining data verification.